test/functional/test_framework/rest.py

```diff
@@ -0,0 +1,23 @@
+import http.client
+
+
+#allows simple http get calls
+def http_get_call(host, port, path, response_object = 0):
```

Comment:
nit: Instead of moving them to a separate file, you could add convenience wrappers to the `test_node` class? Thus, it would not be required to pass in host and port all the time.